packages/office-addin-dev-settings/src/publish.ts

```diff
@@ -0,0 +1,18 @@
+import * as childProcess from "child_process";
+
+/* global console */
+
+export async function publish(zipPath: string): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const command = `npx @microsoft/teamsfx-cli@0.11.0-rc.0 provision manifest --file-path ${zipPath}`;
```

Comment:
Rather than use npx, is there a way to import and use the @microsoft/teamsfx-cli package to do this? It would be more efficient.

Reply:
While I don't have all the details, the way their libraries are designed you have to provide the auth token for methods that end up calling the authenticated endpoint like is happening for this publish.  In order to get the auth prompting they already have built-in, it currently has to go through the cli or the VS Code extension.  Doing npx and using the cli was the simplest option we could get without doing a bunch of auth work ourselves or getting them to rearchitect their libraries to put the pieces together.